### PR TITLE
update dasht-query-line and dasht-query-html to work with tsv

### PR DESCRIPTION
**Pull request by smackesey**
I extracted my work on TSV from the other Dash.app integration fork so that this could be separately merged. It follows your requests in #38.

### CODE REVIEW

File: bin/dasht-query-line [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #
 # ## NAME
 #
-# dasht-query-line - searches [Dash] docsets and emits groups of lines
+# dasht-query-line - searches [Dash] docsets and emits results as tsv
```
Comment (sunaku) [inline bin/dasht-query-line:7]: Please uppercase "TSV".

File: bin/dasht-query-line [hunk 2]
```diff
@@ -28,8 +28,8 @@
 #
 # Searches for *PATTERN* in all installed [Dash] docsets, optionally searching
 # only in those whose names match *DOCSET*s, by calling dasht-query-exec(1)
-# and emits the results in groups of lines, as described in "Results" below.
-# However, if no results were found, this program exits with a nonzero status.
+# and emits the results as TSV. However, if no results were found, this program
```
Comment (sunaku) [inline bin/dasht-query-line:31]: Please spell out "TSV" in prose: "as TSV" -> "in Tab Separated Vector (TSV) format"

File: bin/dasht-query-line [hunk 3]
```diff
@@ -42,26 +42,24 @@
 #
 # ### Results
 #
-# Each search result is printed to stdout as a group of four lines of text:
+# Each search result is printed to stdout as a tab-separated line with fields:
```
Comment (sunaku) [inline bin/dasht-query-line:45]: The fields are separated, not the line; so "as a line with 4 tab-separated fields".

File: bin/dasht-query-line [hunk 4]
```diff
@@ -245,9 +243,11 @@ dasht-docsets "$@" | while read -r docset; do
 
     { $1 = $1 } # strip whitespace from key
 
+    $2 == "=" {
+      result[$1] = substr($0, index($0, $2) + length($2) + 1)
```
Comment (sunaku) [inline bin/dasht-query-line:247]: I would change the `index($0, $2)` call to `length($1) + 1` for robustness (no problem if `$1` contains `=`).

File: bin/dasht-query-line [hunk 5]
```diff
@@ -259,9 +259,12 @@ dasht-docsets "$@" | while read -r docset; do
 
       # resolve URL to filesystem location
       $3 = file_url $3
-    }
 
-    /./ # reject any empty lines from input
+      # print TSV line
```
Comment (sunaku) [inline bin/dasht-query-line:263]: I would drop this (redundant) comment.

File: bin/dasht-query-line [hunk 6]
```diff
@@ -259,9 +259,12 @@ dasht-docsets "$@" | while read -r docset; do
...
+      printf "%s\t%s\t%s\t%s\n", result["name"], docset, \
```
Comment (sunaku) [inline bin/dasht-query-line:264]: I would add parentheses to the `printf()` call and list each argument on its own line.

**Review by sunaku — CHANGES_REQUESTED** [review]
Thanks for this PR.  I've requested some minor changes in the code review.

**Comment by smackesey** [comment]
Made the changes!

**Review by sunaku — CHANGES_REQUESTED** [review]
Thanks @smackesey!  I found some time to experiment with your latest changes and I would suggest applying the changes in the attached patch file in addition to my code review feedback.

* Attachment: [0001-beautify-and-clean-up-documentation.patch.txt](https://github.com/sunaku/dasht/files/7051979/0001-beautify-and-clean-up-documentation.patch.txt)

This PR is coming along nicely and we'll reach the finish line soon! :sweat_smile: Thanks for your patience.

**Review by sunaku — CHANGES_REQUESTED** [review]
Thanks for the update.  There's a potential AWK incompatibility that I've called out in the code review.  No worries about the patch file: I'll apply it on my side when I finally merge the PR. :nerd_face:

**Review by sunaku — CHANGES_REQUESTED** [review]

**Comment by sunaku** [comment]
Looks great, thanks!  I'll try this out for a few days and let you know.  Overall, we should be good to merge. :+1:

**Comment by sunaku** [comment]
Hi @smackesey, I found an issue when searching the BASH docset:

1. Before this PR
```
$ dasht 1 bash
digit-argument (M-0, M-1, … M--)  Bash function
PS1                               Bash variable
self-insert (a, b, A, 1, !, …)    Bash function
```

2. After this PR:
```
$ dasht 1 bash
digit-argument  (M-0, m-1,
PS1              Bash variable
self-insert       (a, b,
```

Following the first result link shows a `Can't load %E2%80%A6` error in w3m.

**Comment by smackesey** [comment]
Fixed, the issue was that the `awk` field separator was still set to the default instead of tab. It didn't affect without spaces in the name, but screwed up those like the ones you posted. Sorry about that.

**Comment by sunaku** [comment]
Sorry for the long delay. :sweat_smile: I'll merge this PR soon (ETA this week).

**Comment by sunaku** [comment]
Merged in commit 882c7f2a6d23aee57d6220eec42a6e101fc8a60a. Thanks for your contribution :pray: and sorry for the long delay! :sweat_smile:

**Pull request closed without merging**